File: Problems 1-9/problem6.py
def sum_of_squares(n = 10):
	sum = 0

	for value in xrange(1, n+1):
		sum += value**2
	
	return sum

def square_of_sums(n = 10):
	square = 0

	for value in xrange(1, n+1):
		square += value
	
	return square**2

def find_difference(n = 100):
	print "Your upper limit is:", n
	print "The sum of squares is:", sum_of_squares(n)
	print "The square of sums is:", square_of_sums(n)
	print "The difference between the above is:", square_of_sums(n) - sum_of_squares(n)

if __name__ == '__main__':
	find_difference()

File: Problems 1-9/problem6.c
#include <stdio.h>

static unsigned long sum_of_squares(unsigned long n);
static unsigned long square_of_sums(unsigned long n);

int main()
{
	unsigned long n = 100;

	printf("Your upper limit is: %lu\n", n);

	printf("The sum of squares is: %lu\n", sum_of_squares(n));
	printf("The square of sums is: %lu\n", square_of_sums(n));

	printf("The difference between the above is: %lu\n", 
		square_of_sums(n) - sum_of_squares(n));

	return 0;
}

unsigned long sum_of_squares(unsigned long n)
{
	unsigned long i, sum = 0;

	for (i = 0; i <= n; i++)
	{
		sum += (i * i);
	}

	return sum;
}

unsigned long square_of_sums(unsigned long n)
{
	unsigned long i, square = 0;

	for (i = 0; i <= n; i++)
	{
		square += i;
	}

	return (square * square);
}
